cmd/snaplock/runinhibit/inhibit.go

```diff
@@ -59,23 +61,78 @@ const (
 	HintInhibitedForPreDownload Hint = "pre-download"
 )
 
+const hintFilePostfix = "lock"
+
 // HintFile returns the full path of the run inhibition lock file for the given snap.
 func HintFile(snapName string) string {
-	return filepath.Join(InhibitDir, snapName+".lock")
+	return filepath.Join(InhibitDir, fmt.Sprintf("%s.%s", snapName, hintFilePostfix))
+}
+
+func InhibitInfoFile(snapName string, hint Hint) string {
+	return filepath.Join(InhibitDir, fmt.Sprintf("%s.%s", snapName, hint))
 }
 
 func openHintFileLock(snapName string) (*osutil.FileLock, error) {
 	return osutil.NewFileLockWithMode(HintFile(snapName), 0644)
 }
 
-// LockWithHint sets a persistent "snap run" inhibition lock, for the given snap, with a given hint.
+// InhibitInfo holds data of the previous snap revision that will be needed by
+// "snap run" while the snap is unlinked (i.e. the current symlink is removed).
+type InhibitInfo struct {
+	// Revision is the previous revision for the snap being refreshed.
+	Revision snap.Revision `json:"revision"`
+}
+
+func validateInhibitInfo(info InhibitInfo) error {
```

Comment:
Could/should this just be a method on `InhibitInfo`? I.e.
```go
func (info InhibitInfo) validate() error {
```

Reply:
updated, thank you!